lib/datadog/appsec/response.rb

```diff
@@ -67,6 +74,37 @@ def content_type(env)
         rescue Datadog::AppSec::Utils::HTTP::MediaRange::ParseError
           DEFAULT_CONTENT_TYPE
         end
+
+        def status
+          Datadog.configuration.appsec.block.status
+        end
+
+        def redirect?
+          REDIRECT_STATUS.include?(status)
+        end
+
+        def location
+          Datadog.configuration.appsec.block.location
+        end
+
+        def content(content_type)
+          setting = Datadog.configuration.appsec.block.templates[content_type]
+
+          case setting
+          when :html, :json, :text
+            Datadog::AppSec::Assets.blocked(format: setting)
+          when String, Pathname
+            path = setting.to_s
+
+            cache[path] ||= (File.open(path, 'rb', &:read) || '')
+          else
+            raise ArgumentError, "unexpected type: #{content_type.inspect}"
+          end
+        end
+
+        def cache
```

Comment:
Can you document in this method what characteristics this cache has?
I see it's in a singleton, so `@cache` will only ever grow, but it looks like it's only based on configuration changes, which require direct user intervention.

I'm suggesting that we write something in code so that we don't have to dig out these answers in the future.